ExampleMod/Projectiles/Minions/ExampleOverheadMinion.cs

```diff
@@ -0,0 +1,84 @@
+            projectile.minion = true; //deals summon damage.
+            projectile.alpha = 90; //slightly transparent.
+            projectile.minionSlots = 1f; //takes up a minion slot.
+        }
+ 
+        public override void Kill(int timeLeft) //on projectile death
+        {
+ 
+            Main.PlaySound(2, projectile.Center, 8); //play a sound 
+ 
+            for (int i = 0; i < 20; i++) 
+            {
+                //spawn dusts on death
+                int dust = Dust.NewDust(new Vector2(projectile.position.X, projectile.position.Y), projectile.width, projectile.height, 16, projectile.velocity.X * 1.2f, projectile.velocity.Y * 1.2f, 160, default(Color), 3.50f);   
+                Main.dust[dust].noGravity = true; //dust is not affected by gravity.
+                Main.dust[dust].velocity *= 2.5f;
+            }
+        }
+ 
+        public override void AI()
+        {       
+            Player player = Main.player[projectile.owner];
+            if (player.dead || !player.active) {
+				player.ClearBuff(ModContent.BuffType<Buffs.ExampleOverheadMinionBuff>()); //handling with the summon buff.
+			}
+			if (player.HasBuff(ModContent.BuffType<Buffs.ExampleOverheadMinionBuff>())) {
+				projectile.timeLeft = 2;
+			}
+            projectile.Center = player.Center - new Vector2(0, 60); //projectile is always above your head.
+            
+            for (int i = 0; i < 200; i++) //this code below handles the aiming and shooting of the minion.
+            {
+                NPC target = Main.npc[i];
+ 
+                
+                float shootToX = target.position.X + (float)target.width * 0.5f - projectile.Center.X; 
+                float shootToY = target.position.Y + (float)target.height * 0.5f - projectile.Center.Y;
+                float distance = (float)System.Math.Sqrt((double)(shootToX * shootToX + shootToY * shootToY));
+ 
+                
+                if (distance < 400f && !target.friendly && target.active && target.CanBeChasedBy()) //range is 400 pixels. 
+                {
+                    if (projectile.ai[0] > 60f) //Fires every second. (60 ticks)
+                    {
+                        
+                        distance = 1.6f / distance;
+ 
+                        
+                        shootToX *= distance * 3;
+                        shootToY *= distance * 3;
+                        int damage = projectile.damage;                  
+                        
+                        Projectile.NewProjectile(projectile.Center.X, projectile.Center.Y, shootToX, shootToY, mod.ProjectileType("ExampleOverheadMinionProjectile"), damage, 4f, Main.myPlayer, 0f, 0f); 
+                        Main.PlaySound(2, (int)projectile.position.X, (int)projectile.position.Y, 28); //28 is the sound
+                        projectile.ai[0] = 0f;
+                    }
+                }
+            }
+            projectile.ai[0] += 1f;
```

Comment:
This code seems oddly similar to lines 57 to 85 from here: https://pastebin.com/1SNQ0sZd and the only difference seems to be that you removed the comments, and changed the distance check and the time between shots, but everything else is the same.
That pastebin is code from an `Al0n37` tutorial (https://www.youtube.com/watch?v=pqb0FKsvKY4)